Feat/material library

ajatdarojat45 opened the pull request:
Hi @SilvinWillemsen,

This PR fixes https://github.com/choras-org/frontend-v2/issues/46.

> Please note that this PR includes code from the previous PR (https://github.com/choras-org/frontend-v2/pull/78 and https://github.com/choras-org/frontend-v2/pull/77), so kindly merge those first.

tigadika committed:
fix: refetch/resuming progress bar when reloading/when selecting the running simulation again

tigadika committed:
fix: auto saved setting when first loaded

tigadika committed:
fix: add material coefficient chart when hovering in select material

tigadika committed:
fix: cleanup sources and receivers when changing model

tigadika committed:
fix: hide simulation button when no simulation available

ajatdarojat45 committed:
Merge pull request #63 from tigadika/fix/viewport

Fix/viewport setting

ajatdarojat45 committed:
chore(release): 0.0.32

ajatdarojat45 committed:
feat: add mixed material detection for surfaces and update UI accordingly

ajatdarojat45 committed:
chore(release): 0.0.33

ajatdarojat45 committed:
Merge pull request #65 from ajatdarojat45/fix/assign-all

Fix/assign all

ajatdarojat45 committed:
feat: refactor CreateMaterialDialog to accept props for open state management

ajatdarojat45 committed:
feat: refactor SurfaceMaterialList to accept props for dialog state management

ajatdarojat45 committed:
feat: enhance SurfacesTab with material library and create material dialog functionality

ajatdarojat45 committed:
chore(release): 0.0.34

SilvinWillemsen commented:
Looking good @ajatdarojat45. A few things here:
- Maybe the "Create new material" button could be renamed to "Create material"? Then the text feels less crammed and it is consistent with the button in the material library.
- The 3-dot button has not yet been removed.
- Could you add "Open material library..." to the material dropdowns? Something along the lines of the image below:

<img width="493" height="429" alt="Screenshot 2025-10-28 at 10 16 42" src="https://github.com/user-attachments/assets/86281857-187b-453f-9a52-b7683abb6b08" />

ajatdarojat45 commented:
Sure, understood @SilvinWillemsen — I’ll update those accordingly.

ajatdarojat45 committed:
refactor: remove unused Button import and comment out Button component in SurfaceMaterialList

ajatdarojat45 committed:
feat: add functionality to open material library from material assignment options

ajatdarojat45 committed:
chore(release): 0.0.35

ajatdarojat45 commented:
Hi @SilvinWillemsen,

I’ve made the changes based on your notes:
- Renamed the button to “Create material” for consistency.
- Removed the 3-dot button.
- For the last point, I’ve added “Open material library...” — but I’ve made a small distinction between the button and the material name to keep it clear.

SilvinWillemsen commented:
Hi @ajatdarojat45, this is all looking good! 

Some very minor changes that are left here:
- Maybe change the title of the "Materials" popup to "Material library" for consistency with the buttons.
- "List of Material Available" can be replaced by "Select a material to view its details" (then this can also be removed from the bottom part of the panel).
- Use a small letter for the 'm' in the "Create material" button in the Material library.
- The distinction you made in the dropdown is good thinking! Perhaps we could leave the font regular (so not italic) and add a simple line divider rather than the inner shadow you seem to use now.

Thanks!

ajatdarojat45 committed:
fix: correct capitalization of "Create material" button text

ajatdarojat45 committed:
fix: update dialog titles for clarity in SurfaceMaterialList

ajatdarojat45 committed:
fix: improve UI consistency by updating "Open material library" text and adding separators

ajatdarojat45 committed:
chore(release): 0.0.36

ajatdarojat45 commented:
Hi @SilvinWillemsen,

I’ve made all the requested changes:
- Updated the popup title to “Material library”.
- Replaced the text with “Select a material to view its details” and removed the redundant one from the bottom.
- Adjusted the “Create material” button text (lowercase “m”).
- Updated the dropdown styling — removed italics and added a simple line divider instead of the inner shadow.

SilvinWillemsen commented:
@ajatdarojat45 Thanks! Looking very good :) The only thing is that the redundant one is still there if no material is selected.

<img width="800" height="710" alt="Screenshot 2025-10-28 at 15 23 28" src="https://github.com/user-attachments/assets/c15fd0ff-3850-4ee0-bc49-64f3f80eb6b4" />

Maybe we can remove the subtitle at the top and leave the bottom one in. What do you think?

ajatdarojat45 commented:
Hi @SilvinWillemsen,

I’d prefer to remove the bottom one and keep the top subtitle, since users usually scan from top to bottom. What do you think?

ajatdarojat45 commented:
Hi @SilvinWillemsen,

Here’s the difference between no material selected and selected material — is this okay?
- No material selected
<img width="1643" height="950" alt="image" src="https://github.com/user-attachments/assets/91adeaa1-90cd-438f-9401-f747fb747759" />

- Material selected
<img width="1644" height="953" alt="image" src="https://github.com/user-attachments/assets/4bfceacd-bd99-475d-9fe5-94c4e7b2c6eb" />

SilvinWillemsen commented:
Yes looking good @ajatdarojat45!

ajatdarojat45 committed:
fix: simplify conditional rendering for selected material in SurfaceMaterialList

ajatdarojat45 committed:
chore(release): 0.0.37

ajatdarojat45 commented:
Thanks, @SilvinWillemsen! Just pushed the latest commit.
